amr-wind/utilities/IOManager.cpp

```diff
@@ -101,6 +107,64 @@ void IOManager::initialize_io()
     }
 }
 
+#ifdef AMR_WIND_USE_ASCENT
+static void ascent_pass(
+    CFDSim& sim,
+    int plt_num_comp,
+    const amrex::Vector<Field*>& plt_fields,
+    const amrex::Vector<std::string>& plt_var_names)
+{
+    // Ascent emit
+    BL_PROFILE("amr-wind::IOManager::ascent");
+
+    amrex::Vector<int> istep(
+        sim.mesh().finestLevel() + 1, sim.time().time_index());
+
+    auto outfield = sim.repo().create_scratch_field(plt_num_comp);
+
+    const int nlevels = sim.repo().num_active_levels();
+
+    for (int lev = 0; lev < nlevels; ++lev) {
+        int icomp = 0;
+        auto& mf = (*outfield)(lev);
+
+        for (auto* fld : plt_fields) {
+            amrex::MultiFab::Copy(
+                mf, (*fld)(lev), 0, icomp, fld->num_comp(), 0);
+            icomp += fld->num_comp();
+        }
+    }
```

Comment:
This makes a third copy of all the output fields which feels unnecessary. Why not just directly pass the `outfield` MultiFab from `write_plot_file` to this function?

Reply:
Good point, I'll explore that approach.